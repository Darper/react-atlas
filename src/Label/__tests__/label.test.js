import React from "react";
import { mount } from "enzyme";
import { Label } from "../index";
import renderer from "react-test-renderer";

describe("Test Label component", () => {
  it("Test render correctly", () => {
    const comp = <Label htmlFor="someField" label="Some Field:" required />;
    const tree = renderer.create(comp).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("htmlFor renders for attribute", () => {
    const label = mount(<Label htmlFor="someField" label="Some Field:" />);

    expect(label.props().htmlFor).toBe("someField");
    expect(
      label
        .find("label")
        .find('[htmlFor="someField"]')
        .exists()
    ).toBe(true);
  });

  it("inline && leftLabel renders with inline css classes", () => {
    const label = mount(
      <Label htmlFor="someField" label="Some Field:" inline leftLabel />
    );

    expect(label.find(".inline").exists()).toBe(true);
  });

  it("renders label text", () => {
    const label = mount(
      <Label htmlFor="someField" label="Some Field:" inline />
    );

    expect(label.find("label").text()).toEqual("Some Field:");
  });

  it("leftLabel omits unneeded css classes", () => {
    const label = mount(
      <Label htmlFor="someField" label="Some Field:" leftLabel />
    );

    expect(label.find(".labelSpacing").exists()).toBe(false);
    expect(label.find(".verticalPadding").exists()).toBe(false);
  });

  it("renders * by default for required", () => {
    const label = mount(
      <Label htmlFor="someField" label="Some Field:" required />
    );

    expect(label.find(".required").html()).toContain("*");
  });

  it("renders text if supplied for required", () => {
    const label = mount(
      <Label
        htmlFor="someField"
        label="Some Field:"
        required="this field is required"
      />
    );

    expect(label.find(".required").html()).toContain("this field is required");
  });

  it("renders text if supplied for required", () => {
    const label = mount(
      <Label
        htmlFor="someField"
        label="Some Field:"
        required="this field is required"
      />
    );

    expect(label.find(".required").html()).toContain("this field is required");
  });

  it("applies error css class if status is error and required is true", () => {
    const label = mount(
      <Label htmlFor="someField" label="Some Field:" status="error" required />
    );

    expect(label.find(".required_error").exists()).toBe(true);
  });
});
