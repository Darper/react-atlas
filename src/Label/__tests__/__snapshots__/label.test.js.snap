// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Label component Test render correctly 1`] = `
<div
  className="tooltipRight labelContainer"
>
  <div
    className="verticalPadding horizontalPadding"
  >
    <label
      className="labelSpacing label label"
      htmlFor="someField"
      style={undefined}
    >
      Some Field:
    </label>
    <span
      className="required"
    >
       
      <span
        className="defaultFont"
      >
        *
      </span>
    </span>
  </div>
</div>
`;
